library/cc/engine_builder.cc

```diff
@@ -202,27 +202,6 @@ EngineBuilder::enablePlatformCertificatesValidation(bool platform_certificates_v
 }
 
 std::string EngineBuilder::generateConfigStr() const {
-#if defined(__APPLE__)
```

Comment:
If I'm following the PR and your comment correctly, I think the reason we'll use the Apple resolver without this block regardless of the Builder type (after this PR lands) is because of the change to config.cc:

https://github.com/envoyproxy/envoy-mobile/pull/2618/files#diff-1af53ae37f8c227f37b80bcb6e2335f55b2f830aa658ac77e099f2549f157712

Unless there is code to specifically override `dns_resolver_config` (which there shouldn't be since @Augustyniak seems to have removed it in this PR), it will use the Apple resolver on iOS and getaddrinfo elsewhere. Does that sound right to you?

Reply:
fair enough.  the one other thing i'd ask for is a follow-up tracking issue on removing the cares dependency, which may be an upstream thing

Reply:
>Unless there is code to specifically override dns_resolver_config (which there shouldn't be since @Augustyniak seems to have removed it in this PR), it will use the Apple resolver on iOS and getaddrinfo elsewhere. Does that sound right to you?

1. iOS engine builder always uses `apple dns resolver`
2. Kotlin engine builder uses `use apple dns resolver` on mac and `getaddrinfo` on linux. I can modify it so that it always uses `getaddtinfo` - let me know.
3. c++ engine builder uses `apple dns resolver` on mac/OS and `getaddrinfo` resolver everywhere else.

I would appreciate your thoughts on 2). Will look into @alyssawilk 's [comment more](https://github.com/envoyproxy/envoy-mobile/pull/2618#discussion_r1000831843) and open GH issue if I cannot figure it out.

Reply:
opened https://github.com/envoyproxy/envoy-mobile/issues/2629